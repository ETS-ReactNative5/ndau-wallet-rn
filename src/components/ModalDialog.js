import React, { Component } from 'react';

import { StyleSheet, Modal, View, TouchableHighlight, Text } from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import styleConstants from '../css/styleConstants';

class ModalDialog extends Component {
  render() {
    return (
      <Modal animationType="slide" transparent={true} onRequestClose={() => {}} {...this.props}>
        <View style={[ styles.outerView, this.props.outerViewStyle ]}>
          <View style={[ styles.innerView, this.props.innerViewStyle ]}>
            <TouchableHighlight
              onPress={() => {
                this.props.setModalVisible(false);
              }}
            >
              <FontAwesome
                style={styles.closeButton}
                name="close"
                color={styleConstants.ICON_GRAY}
                size={20}
              />
            </TouchableHighlight>
            {this.props.children}
          </View>
        </View>
      </Modal>
    );
  }
}

var styles = StyleSheet.create({
  outerView: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  innerView: {
    backgroundColor: '#333333',
    borderRadius: 3,
    width: wp('90%'),
    height: hp('70%'),
    paddingLeft: wp('2%'),
    paddingRight: wp('2%'),
    paddingBottom: hp('2%')
  },
  closeButton: {
    marginTop: hp('.5%'),
    marginLeft: wp('1%')
  }
});

export default ModalDialog;
