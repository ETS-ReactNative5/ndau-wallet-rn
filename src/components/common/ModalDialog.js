import React, { Component } from 'react'
import { StyleSheet, Modal, View, TouchableHighlight, Text } from 'react-native'
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp
} from 'react-native-responsive-screen'
import FontAwesome5Pro from 'react-native-vector-icons/FontAwesome5Pro'
import Padding from './Padding'
import styleConstants from '../../css/styleConstants'

class ModalDialog extends Component {
  constructor (props) {
    super(props)

    this.state = {
      visible: false
    }
  }

  showModal = () => {
    this.setState({ visible: true })
  }

  closeModal = () => {
    this.setState({ visible: false })
  }

  render () {
    return (
      <Modal
        animationType='slide'
        transparent
        onRequestClose={() => {}}
        visible={this.state.visible}
        {...this.props}
      >
        <View style={[styles.outerView, this.props.outerViewStyle]}>
          <View style={[styles.innerView, this.props.innerViewStyle]}>
            <Padding top={0.5}>
              <TouchableHighlight onPress={this.closeModal}>
                <FontAwesome5Pro
                  style={styles.closeButton}
                  name='times'
                  color={styleConstants.ICON_GRAY}
                  size={20}
                  light
                />
              </TouchableHighlight>
            </Padding>
            {this.props.children}
          </View>
        </View>
      </Modal>
    )
  }
}

var styles = StyleSheet.create({
  outerView: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  innerView: {
    backgroundColor: '#0F2748',
    borderRadius: 3,
    width: wp('90%'),
    height: hp('60%'),
    paddingLeft: wp('3%'),
    paddingRight: wp('3%'),
    marginTop: hp('20%')
  },
  closeButton: {}
})

export default ModalDialog
