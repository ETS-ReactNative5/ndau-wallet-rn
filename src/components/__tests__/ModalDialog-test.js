import { mount } from 'enzyme'
import React from 'react'
import { StyleSheet } from 'react-native'
import ModalDialog from '../ModalDialog'

import renderer from 'react-test-renderer'

describe('testing ModalDialog...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  })
  const navigator = {
    setStyle: () => {}
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<ModalDialog parentStyles={styles} navigator={navigator} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
