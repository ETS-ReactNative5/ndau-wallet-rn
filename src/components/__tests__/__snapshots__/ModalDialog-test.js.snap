// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing ModalDialog... renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  navigator={
    Object {
      "setStyle": [Function],
    }
  }
  onRequestClose={[Function]}
  parentStyles={
    Object {
      "wizardText": Object {
        "color": "#ffffff",
        "fontSize": 20,
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#333333",
            "borderRadius": 3,
            "height": 934,
            "paddingBottom": 26.5,
            "paddingLeft": 15,
            "paddingRight": 15,
            "width": 675,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#A9A9A9",
                "fontSize": 20,
              },
              Object {
                "marginLeft": 7.5,
                "marginTop": 6.5,
              },
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          ÔÄç
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
